### PR TITLE
[jsforce] add streaming extensions

### PR DESCRIPTION
**Pull request by gidsg**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jsforce/jsforce/pull/740

CC @mars this is my best attempt, although I am new to TypeScript it does compile for my code and your example code. I suspect it may be possible to make it more specific.

### CODE REVIEW

**Comment by dolanmiu** [comment]
Can you address those concerns I mentioned?

**Comment by andrewbranch** [comment]
@dolanmiu if you have a chance, would you mind requesting changes via the GitHub review feature so this PR gets its status updated accordingly? That will expedite the process of getting the changes re-reviewed after @gidsg makes them.

**Comment by gidsg** [comment]
> Can you address those concerns I mentioned?

@dolanmiu It doesn't look like you have made any comments on this pull request prior to this. Perhaps you had them in an in-progress review and did not submit it?

**Comment by dolanmiu** [comment]
There

File: types/jsforce/streaming.d.ts [hunk 1]
```diff
@@ -18,6 +18,16 @@ export class Streaming extends EventEmitter {
...
 }
+
+export namespace StreamingExtension {
+    export class Replay {
+        constructor(channel: string, replayId: number)
+    }
+    export class AuthFailure {
+               constructor(failureCallback: () => any);
+           }
+}
```
[the remark this answers is not in the document]
Comment (dolanmiu) [reply types/jsforce/streaming.d.ts]: (and press format too)

**Comment by gidsg** [comment]
@dolanmiu I have addressed your comments now, please take another look.

**Comment by mars** [comment]
Thanks for the @ mention, but I do not know much about TypeScript, so I'm not going to be helpful determining what the types should be.

**Comment by borys-kupar** [comment]
> Can I be removed from this package? I no longer work with salesforce and it's a bit tiring receiving all of these messages for jsforce — Abe White
> […](#)
> On Oct 16, 2019, at 07:15, Gideon Goldberg ***@***.***> wrote: ﻿ @dolanmiu I have addressed your comments now, please take another look. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

@WhiteAbeLincoln sure, just submit a PR to remove yourself from maintainers list: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jsforce/index.d.ts

**Pull request merged**